packages/core/src/uma.ts

```diff
@@ -654,11 +661,14 @@ export async function getPayReqResponse({
     ? Math.round((request.amount - receiverFeesMillisats) / conversionRate)
     : request.amount;
 
+  const encodedInvoiceUUID =
+    request.invoiceUUID && JSON.stringify(request.invoiceUUID);
+
   const encodedPayerData =
     request.payerData && JSON.stringify(request.payerData);
   const encodedInvoice = await invoiceCreator.createUmaInvoice(
     msatsAmount,
-    metadata + (encodedPayerData || ""),
+    metadata + (encodedPayerData || "") + (encodedInvoiceUUID || ""),
```

Comment:
@zhenlu does encodedPayerData also need to be added this way?

Reply:
No, I think it's like this all across the SDKs.

Reply:
I feel like we should do it as this, but since it is already released so that might be complicated to do. cc @jklein24

Reply:
I'm not super worried about breaking changes for invoices at the moment since no one is using them yet. If we need to make changes you guys think are worth making, I say go for it.